Components/Motors/CytronMD30C.cs

```diff
@@ -2,67 +2,108 @@
 using Scarlet.Filters;
 using Scarlet.IO;
 using Scarlet.Utilities;
+using System.Threading;
 
 namespace Scarlet.Components.Motors
 {
     public class CytronMD30C : IMotor
     {
-        private IFilter<float> Filter;
+        private IFilter<float> Filter; // Filter for speed output
         private readonly IPWMOutput PWMOut;
         private readonly float MaxSpeed;
         private readonly IDigitalOut GPIOOut;
 
-        private bool Stopped;
-        public float TargetSpeed { get; private set; }
-
+        private bool Stopped; // Whether or not the motor is stopped
+        private bool OngoingSpeedThread; // Whether or not a thread is running to set the speed
+        public float TargetSpeed { get; private set; } // Target speed (-1.0 to 1.0) of the motor
 
+        /// <summary> Initializes a CytronMD30C Motor controller </summary>
+        /// <param name="PWMOut"> PWM Output to set the speed of the motor controller </param>
+        /// <param name="GPIOOut"> GPIO Output that sets the direction of the motor </param>
+        /// <param name="MaxSpeed"> Limiting factor for speed (should never exceed + or - this val) </param>
+        /// <param name="SpeedFilter"> Filter to use with MC. Good for ramp-up protection and other applications </param>
         public CytronMD30C(IPWMOutput PWMOut, IDigitalOut GPIOOut, float MaxSpeed, IFilter<float> SpeedFilter = null)
         {
             this.PWMOut = PWMOut;
-            this.MaxSpeed = MaxSpeed;
+            this.MaxSpeed = Math.Abs(MaxSpeed);
             this.Filter = SpeedFilter;
             this.PWMOut.SetFrequency(5000);
             this.PWMOut.SetEnabled(true);
             this.GPIOOut = GPIOOut;
             this.GPIOOut.SetOutput(false);
+            this.SetSpeedDirectly(0.0f);
         }
 
-        public void EventTriggered(object Sender, EventArgs Event)
-        {
+        public void EventTriggered(object Sender, EventArgs Event) { }
 
+        /// <summary> 
+        /// Immediately sets the enabled status of the motor. 
+        /// If false, resets TargetSpeed to 0 and stops the motor.
+        /// </summary>
+        public void SetEnabled(bool Enabled)
+        {
+            this.Stopped = !Enabled;
+            if (!Enabled)
+            {
+                this.TargetSpeed = 0;
+                this.SetSpeed(0);
```

Comment:
Shouldn't this be `this.SetSpeedDirectly(0);`?

Reply:
Yes, I will fix that.